build/Jenkinsfile.pr

```diff
@@ -358,6 +359,173 @@ pipeline {
 				}
 
 
+				dir('src/github.com/mattermost/mattermost-server') {
+					ansiColor('xterm') {
+						sh """
+						echo "Ignoring known MySQL mismatch: ChannelMembers.SchemeGuest"
+						/usr/local/bin/docker-compose --no-ansi -f build/docker-compose.yml exec -T mysql mysql -D migrated -uroot -pmostest -e "ALTER TABLE ChannelMembers DROP COLUMN SchemeGuest;"
+						/usr/local/bin/docker-compose --no-ansi -f build/docker-compose.yml exec -T mysql mysql -D latest -uroot -pmostest -e "ALTER TABLE ChannelMembers DROP COLUMN SchemeGuest;"
+
+						echo "Generating dump"
+						/usr/local/bin/docker-compose --no-ansi -f build/docker-compose.yml  exec -T mysql mysqldump --skip-opt --no-data --compact -u root -pmostest migrated > migrated.sql
+						/usr/local/bin/docker-compose --no-ansi -f build/docker-compose.yml  exec -T mysql mysqldump --skip-opt --no-data --compact -u root -pmostest latest > latest.sql
+
+						echo "Removing databases created for db comparison"
+						/usr/local/bin/docker-compose --no-ansi -f build/docker-compose.yml  exec -T mysql mysql -uroot -pmostest -e \"DROP DATABASE migrated; DROP DATABASE latest\"
+
+						echo "Generating diff"
+						diff migrated.sql latest.sql > diff.txt
+						export diffErrorCode=\$?
+
+						if [ \$diffErrorCode -eq 0 ]; then echo \"Both schemas are same\";else cat diff.txt; fi
+
+						exit \$diffErrorCode
+						"""
+					}
+				}
+
+				withDockerContainer(args: "-u root --privileged --net ${COMPOSE_PROJECT_NAME}_mm-test -v ${WORKSPACE}/src:/go/src/", image: 'mattermost/mattermost-build-server:oct-18-2019') {
+					ansiColor('xterm') {
+						sh """
+							cd /go/src/github.com/mattermost/mattermost-server
+
+							mkdir -p client/plugins
+							cat config/config.json
+
+							make test-server BUILD_NUMBER='${BRANCH_NAME}-${BUILD_NUMBER}' TESTFLAGS= TESTFLAGSEE=
+						"""
+					}
+					withCredentials([string(credentialsId: 'CODECOV_TOKEN', variable: 'CODECOV')]) {
+						sh '''
+							cd /go/src/github.com/mattermost/mattermost-server
+							curl -s https://codecov.io/bash | bash -s - -t $CODECOV || echo 'Codecov failed to upload'
+						'''
+					}
+				}
+			}
+		}
+
+		stage('Test-Postgres') {
```

Comment:
@metanerd @cpanato This is almost a copy and paste of the `Test-Mysql` (so must exist a better way to do it with less repeated code) and probably we would be able to do it in parallel, but I'm not very familiarized with this config files, so any guidance here is very welcome.

Reply:
`stage('Test Databases') {
			parallel {
				stage('Test-Mysql') {`

The problem is the same docker setup with ports already in use. So we should untangle that if we want to run it in parallel in the future. Anyway, this PR looks like a huge improvement!